### DISCUSSION TITLE
Text-bison-001 model providing special characters in model response

### DISCUSSION THREAD

**Discussion by adityaworkfusion** [GoogleCloudPlatform/generative-ai#246] [General]
Hi,

I need help here.

So, I have used the text bison model in my project, I am using chroma embeddings first to store the input documents, later with the similarity, appropriate embeddings will get called and at the end this will provide as context along with the prompt to the text-bison model.

Now here text bison model generates special characters for some responses. Can someone help here to resolve this.

Thanks
Aditya

**Comment by adityaworkfusion** [comment]
@polong-lin can you help here.

**Reply by polong-lin** [reply]
Hi @adityaworkfusion , could you clarify what you mean by:
"Now here text bison model generates special characters for some responses. "

It sounds like you're doing retrieval augmented generation using ChromaDB as your vector store, and perhaps text-bison is generating characters that are unexpected? What are some examples of what you did not want to see, and what you expected instead? 

Could you perhaps just do some post-processing, if there are unwanted characters?

**Comment by adityaworkfusion** [comment]
Here I have attached one response, I have highlighted the errors.

**Reply by adityaworkfusion** [reply]
Example Response - The 20â€“30 minutes that analysts had spent manually researching an entity is easily cut to 2â€“3 minutes as Evelyn collects, analyzes, and reports on her findings. Whatâ€™s even better is that Evelynâ€™s performance improves over time. Like all WorkFusion Digital Workers, her AI core continuously learns with every assignment and interaction â€” delivering powerful results for reducing manual review time, false-positive alerts, and compliance risk."

**Reply by polong-lin** [reply]
Ah I'm not sure why it's generating certain symbols in that encoding, but you can do some post-processing to fix the encoding. For example:
https://stackoverflow.com/questions/55737316/python-selenium-text-returns-%C3%A2%E2%82%AC-instead-of-apostrophe

**Reply by adityaworkfusion** [reply]
Just to add on to above query - We experience sometime the model respond with raw text, rather than providing the proper format and font of text in response, it provides the same text as it has been embedded or trained on.

 How to handle such scenarios @polong-lin
